File: swift/Sources/Core/Storage.swift
import Foundation

final class Storage<T: IdentifiableModel> {
    private var items: [T] = []
    
    func add (_ element : T) {
        if let idx = items.firstIndex(where: { $0.id == element.id }) {
            items[idx] = element
        } else {
            items.append(element)
        }
    }
    
    func remove(id: String) {
        items.removeAll { $0.id == id }
    }
    
    func getAll() -> [T] {
        return items
    }
    
    func value(by id: String) -> T? {
        items.first { $0.id == id }
    }
}

File: kotlin/src/main/kotlin/Storage.kt
class Storage<T : Identifiable> {
    private val items = mutableListOf<T>()

    fun add(element: T) {
        val idx = items.indexOfFirst { it.id == element.id }
        if (idx >= 0) items[idx] = element else items.add(element)
    }

    fun remove(id: String) {
        items.removeAll { it.id == id }
    }

    fun getAll(): List<T> = items.toList()

    fun valueById(id: String): T? = items.firstOrNull { it.id == id }
}
